mesonbuild/backend/backends.py

```diff
@@ -515,6 +515,10 @@ def write_test_serialisation(self, tests, datafile):
                 self.environment.cross_info.need_exe_wrapper()
             if isinstance(exe, build.BuildTarget):
                 is_cross = is_cross and exe.is_cross
+            if isinstance(exe, dependencies.ExternalProgram):
+                # E.g. an external verificator or simulator program run on a generated executable.
+                # Can always be run.
+                is_cross = False
```

Comment:
Won't this conflict with #2220?

Reply:
The correct name for this would be `is_cross_built`, because we have two different meanings for `is_cross`. Probably need to do some renaming.